### PR TITLE
Re-use existing SSH connection

### PR DESCRIPTION
**Pull request by mschnitzer**
Squashes #1276 

Opening multiple ssh connection could accidentally be considered as:
- ssh brute-force or
- some other kind of malicious network activity

This does also improve the speed of calling remote commands.

### CODE REVIEW

**Comment by tgoettlicher** [comment]
Have you verified that it works with servers running old versions of sshd?

**Comment by mschnitzer** [comment]
@tgoettlicher That's a good point, thanks. I'll double check that tomorrow.

File: lib/remote_system.rb [hunk 1]
```diff
@@ -19,6 +19,9 @@ class RemoteSystem < System
   attr_accessor :host
   attr_accessor :remote_user
 
+  SSH_BASIC_OPTIONS = ["-o", "ControlMaster=auto", "-o",
+    "ControlPath=~/.ssh/socket%r@%h-%p", "-o", "ControlPersist=600"]
```
Comment (tgoettlicher) [inline lib/remote_system.rb]: `ssh` will fail if the `~/.ssh` directory doesn't exists.

**Comment by mschnitzer** [comment]
@cornelius we should have a discussion in a meeting about that.

**Pull request closed without merging**